recursion.rb

```diff
@@ -1,14 +1,43 @@
 
 def fact(n)
+  # RC:  n! = n * (n-1)!
+  # BC:  1! = 1
+  # BC:  0! = 1
+  return 1 if n == 1 || n == 0
+  n * fact(n-1)
 end
 
 def fib(n)
+  # RC: n + (n-1)
+  # BC: 4
+  # 1, 1, 2, 3, 5
+  return n if n == 0 || n == 1
+  fib(n-1) + fib(n-2)
+
 end
 
 def pal(s)
+  pal_helper(s, 0,-1, true)
+end
+
+def pal_helper(s,x, y, result)
+  return result if x == s.length/2
+  result = s[x]==s[y]
+  result = result&&pal_helper(s, x+1, y-1, result)
 end
```

Comment:
Good solution. Interesting use of a helper function here. Note that your function will continue all the way through a particularly long string, even if it "proves" it's not a palindrome on the first characters. Check some other solutions for ones that return early. (That efficiency isn't too important, though, and yours works just fine.)